server/src/test/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainIT.java

```diff
@@ -1037,8 +1036,21 @@ public void testCannotAllocateStaleReplicaExplanation() throws Exception {
         logger.info("--> stop node with the replica shard");
         internalCluster().stopRandomNode(InternalTestCluster.nameFilter(replicaNode));
 
-        logger.info("--> index more data, now the replica is stale");
-        indexData();
+        final IndexMetaData.State indexState = randomIndexState();
+        if (indexState == IndexMetaData.State.OPEN) {
+            logger.info("--> index more data, now the replica is stale");
+            indexData();
+        } else {
+            logger.info("--> close the index, now the replica is stale");
+            assertAcked(client().admin().indices().prepareClose("idx"));
```

Comment:
will this turn cluster health RED? If so, I think that that is not desired behavior. Let's add a bullet point to the meta-issue that we will change that.

Reply:
It does not turn the cluster health red. Here we use the cluster health to check for index health after the close so that there's no initializing shards. I've added a missing `setWaitForNoInitializingShards`.